rkt/metadata_service.go

```diff
@@ -429,13 +447,21 @@ func handleImageManifest(w http.ResponseWriter, r *http.Request, _ *schema.PodMa
 func handleAppID(w http.ResponseWriter, r *http.Request, pm *schema.PodManifest, im *schema.ImageManifest) {
 	defer r.Body.Close()
 
+	n := mux.Vars(r)["app"]
+	an, err := types.NewACName(n)
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		fmt.Fprintf(w, "App name %q is not a valid AC Name", n)
+		return
+	}
+
 	w.Header().Add("Content-Type", "text/plain")
 	w.WriteHeader(http.StatusOK)
-	a := pm.Apps.Get(im.Name)
-	if a == nil {
-		panic("could not find app in manifest!")
+	app := pm.Apps.Get(*an)
+	if app == nil {
+		panic("impossible: could not find app in manifest!")
```

Comment:
yes, because we have checked that the image manifest is not nil in `appGet`, so here in the podManifest, the RuntimeApp cannot be nil